code/game/gamemodes/changeling/powers/transform.dm

```diff
@@ -16,12 +16,15 @@
 		return
 	user.dna = chosen_dna.Clone()
 	user.real_name = chosen_dna.real_name
-	user.flavor_text = ""
 	if(ishuman(user))
-		user.set_species()
+		user.set_species(chosen_dna.species)
+	domutcheck(user, null, MUTCHK_FORCED) //Ensures species that get powers by the species proc handle_dna keep them
+	user.flavor_text = ""
+	user.dna.UpdateSE()
+	user.dna.UpdateUI()
+	user.sync_organ_dna(1)
 	user.UpdateAppearance()
-	domutcheck(user, null)
-	
+
```

Comment:
Aaaand these.

As I recall, these needed to be added in to make the colorblindness gene behave properly, but that shouldn't be an issue now that it functions like the other disabilities.

If they're unrelated fixes, it'd be preferable to have them in a separate PR that explains exactly what they do, instead of quietly fixing them in a PR with no reason to touch them.